kclvm/driver/src/kpm_update.rs

```diff
@@ -0,0 +1,57 @@
+use crate::kpm_metadata::get_path_for_executable;
+use anyhow::{bail, Result};
+use std::{path::PathBuf, process::Command};
+
+const MANIFEST_FILE: &str = "kcl.mod";
+
+pub(crate) fn update_kcl_module(manifest_path: PathBuf) -> Result<()> {
+    match lookup_the_nearest_file_dir(manifest_path.clone(), MANIFEST_FILE) {
+        Some(mod_dir) => {
+            match Command::new(kcl())
+                .arg("mod")
+                .arg("update")
+                .current_dir(mod_dir)
+                .output()
+            {
+                Ok(output) => {
+                    if !output.status.success() {
+                        bail!(
+                            "update failed with error: {}",
+                            String::from_utf8_lossy(&output.stderr)
+                        );
+                    }
+                    Ok(())
+                }
+                Err(err) => bail!("update failed with error: {}", err),
+            }
+        }
+        None => bail!(
+            "Manifest file '{}' not found in directory hierarchy",
+            MANIFEST_FILE
+        ),
+    }
+}
+pub fn kcl() -> PathBuf {
```

Comment:
Why are these functions defined repeatedly?

Reply:
Fixed it,<mark> kcl </mark> executable is required over there and idk why it was failing without describing it separately again in kpm_update.rs, removed kpm() as it had no use over here